### ISSUE TITLE
Modest Swift 5.1 updates needed

### ISSUE THREAD

**Issue by rayfix** [raywenderlich/swift-style-guide#314]
A couple of things:

1. Single expressions of return should elide the `return` keyword.  I think this keeps with the spirit of minimalism in the current guide and is what the Swift Apprentice and Swift Algorithms books have adopted in the coming new editions.

2. We should probably solicit feedback from the SwiftUI book authors for best practices on formatting body function builders.

**Comment by ondrejhanslik** [comment]
I am not sure avoiding explicit `return` is a good advice in general. I would recommend the same I am using in other languages - avoid `return` only in single-line closures/functions. Not in single-expressions.

e.g. 

    override var myVar: Int { 10 }

but

   override var myVar: Int {
      return 10
   }

or

    let element = array.first { $0.isValid }

but

       let element = array.first {
          return $0.isValid 
       }

In my opinion, that's the best for readability. I am reluctant to omit a keyword just because it doesn't have to be there.

**Comment by hollance** [comment]
```
override var myVar: Int { 10 }
```

This is not a useful example, in my opinion, because you wouldn't often write that in a real program.

While the second example is not new; we've always been allowed to drop `return` in closures and I don't think the multi-line version with `return` is very common.

**Comment by marinbenc** [comment]
I think we also need to clarify, when writing a SwiftUI `body`, which method calls should be indented and when. For instance:

```swift
Button(action: buttonPressed) {
  Text("Hello!")
    .padding()
}.padding()
```

vs 

```swift
Button(action: buttonPressed) {
  Text("Hello!")
    .padding()
}
.padding()
```

Likewise, I'd suggest we advise to always split button actions into specific methods, so they don't clutter the `body`.

There should probably be a lot more SwiftUI specific standards.

**Comment by rcritz** [comment]
@marinbenc tell me more about your feeling on splitting button actions into specific methods. The idiom seems to be to put a closure there, but that will flag with SwiftLint with the raywenderlich.com ruleset. Doing as you suggest gets rid of the flagging, which is good. But I hesitate to make that a style guide requirement.

Oh, and the second form you show above is the correct form. The first will flag.

**Comment by rayfix** [comment]
@rcritz @marinbenc Hi! Multiple trailing closure syntax is coming in Swift 5.3 which will undoubtedly be released during WWDC20.   I think it will look something like this:

```swift
Button {
  Text("Hello!").padding()
} action: {
   print("Hello tapped!")
}
.padding()
```

Notably this has been added to the API Design Guidelines:

> Name functions assuming that the argument label of the first trailing closure will be dropped. Include meaningful argument labels for all subsequent trailing closures.

See:

 https://github.com/apple/swift-evolution/blob/master/proposals/0279-multiple-trailing-closures.md

**Comment by marinbenc** [comment]
I agree that probably moving functions out of `body` and into methods is more of a case-by-case type of thing, so probably shouldn't be enforced by the style guide.

**Comment by NikKovIos** [comment]
I disagree with excluding return word. Understanding is preferable that minimalism!
